src/ape/contracts/base.py

```diff
@@ -663,6 +665,8 @@ class ContractInstance(BaseAddress, ContractTypeWrapper):
         contract = a.deploy(project.MyContract)  # The result of 'deploy()' is a ContractInstance
     """
 
+    _default_sender: Optional[AccountAPI] = None
```

Comment:
This should not be on a contract instance.
Instead, there should be a global, ape-wide list of default sender addresses so that we can support nested context managers:

```python
 with ape.default_sender(foo):
   ... (uses foo)
   with ape.default_sender(bar):
   .... (uses bar)
     with ape.default_sender(None):
      .... (make sure there is no default sender)
```